### ISSUE TITLE
Microsoft Owin OpenIdConnect setting cookie behind in time only in chrome

### ISSUE THREAD

**Issue by SpeakOutLoud** [aspnet/AspNetKatana#370]
Hi,

As of  chrome update **84.0.4147.105** (version 105) the use of 'UseOpenIdConnectAuthentication' in AspNet MVC 4.5.2 Net Framework is setting the cookie OpenIdConnect.nonce... with 'Expire' behind in time.

Currently I am using Microsoft.Owin 4.1 and other pakcages of Owin of 4.1

I have modified the cookie with ICookieManager and set the 'Expire' cookie to the future, but for some reason when the IdentityServer4 returns the the client, the client is unable to find the cookie OpenIdConnect.nonce, therefor is throwing nonce not found (check the Fig1. below). I have checked with fiddler to see if the cookie is and even with debugging in visual studio, in both cases the cookie was set.

Fig1
https://imgur.com/nuIR4cY

Thank you

**Comment by Tratcher** [comment]
Please share a decrypted Fiddler trace file showing the whole flow before your modifications.

It's standard to set Expire in the the past when deleting a cookie.

**Comment by SpeakOutLoud** [comment]
Hi Tratcher, sorry for the late response!

I have here 2 images that, the fist shows when the client is called and redirects the user to IDP, fiddler trace shows the cookie being set and the date is being behind in time. 
And the 2nd image is, shows when the client is recieving the response from the IDP and is requesting the cookie. but the value is null.

I tried copying the value of the cookie OpenIdConnect.nonce... and paste the value on debug when client requests the cookie, and it worked like a charm, but when it returns null it throws "IDX21323: OpenIdConnectProtocolValidationContext.Nonce was null,"

![image](https://user-images.githubusercontent.com/43482873/89166326-2e6f6480-d57a-11ea-95aa-ddae1f30aa0c.png)

![image](https://user-images.githubusercontent.com/43482873/89166422-4d6df680-d57a-11ea-993c-19c79c6f06d5.png)

**Comment by Tratcher** [comment]
Note the GMT time zone. Is your time zone UTC+3? Also compare this value to to Date response header.

I think your issue is not the time, but that the cookie is missing the "Secure" attribute. That's required when using SameSite=None. It should have been added automatically if the request was https, unless you're using a proxy?

**Comment by SpeakOutLoud** [comment]
Hey Trachter, 
Thank you for pointing me in the right direction. At least I found a solution to this case, forcing the cookie to be same site and secure.

![image](https://user-images.githubusercontent.com/43482873/89535250-75698e00-d7f6-11ea-8597-bfc79cb4d00c.png)

But as I said before, the chrome update 84.0.4147.105 (version 105) had made this change,  as before the update the app used to work in production without any issues.

Thank you again! 
Closing this.
